fiona/ogrext.pyx

```diff
@@ -775,6 +775,9 @@ cdef class Session:
             raise ValueError("Null layer")
 
         result = OGR_L_GetExtent(self.cogr_layer, &extent, 1)
+        
+        if result != OGRERR_NONE:
+            raise DriverError("Driver was not able to calculate extent")
```

Comment:
maybe "Driver was not able to calculate bounds" would be better, as Collection has a bounds property and not an extent property?

Reply:
I agree that "bounds" would be more useful as that is the terminology we expose to the user.